src/slic3r/GUI/PhysicalPrinterDialog.cpp

```diff
@@ -311,6 +334,41 @@ void PhysicalPrinterDialog::build_printhost_settings(ConfigOptionsGroup* m_optgr
     update();
 }
 
+void PhysicalPrinterDialog::update_ports() {
+    const PrinterTechnology tech = Preset::printer_technology(*m_config);
+    if (tech == ptFFF) {
+        const auto opt = m_config->option<ConfigOptionEnum<PrintHostType>>("host_type");
+        if (opt->value == htObico) {
```

Comment:
Can we move this part to Obico.cpp or CloudPrinterAuthDialog.cpp? We want to minimize the amount of code specific to Obico in the common classes